### ISSUE TITLE
Docker scan on a non-local image fails but does download image

### ISSUE THREAD

**Issue by grahamlyons** [snyk/snyk#595]
- `node -v`: n/a
- `npm -v`: n/a
- `snyk -v`: 1.180.1
- Command run: `snyk test`

### Expected behaviour
Run Snyk against a non-local Docker image and _either_ the image is downloaded and tested *or* Snyk reports that the image can't be found.

### Actual behaviour
Snyk downloads the image and then reports that it can't be found.

### Steps to reproduce
 - Run `snyk test` on a Docker image that is not found locally
 - Without pulling the image down yourself, run `snyk test` again on the same image

```bash
$ alias snyk='docker container run --rm -e SNYK_TOKEN -v "$(pwd)":/project -v /var/run/docker.sock:/var/run/docker.sock snyk/snyk-cli:docker'

$ snyk test --docker mergermarket/cdflow-commands:153 --file=Dockerfile
Failed to run the process ...

Testing mergermarket/cdflow-commands:153...

Docker error: image was not found locally: mergermarket/cdflow-commands:153

$ docker image ls 
REPOSITORY                     TAG                 IMAGE ID            CREATED             SIZE
snyk/snyk-cli                  docker              bbdf7e98a34d        26 hours ago        810MB
mergermarket/cdflow-commands   153                 0e510459f991        4 weeks ago         679MB

$ snyk test --docker mergermarket/cdflow-commands:153 --file=Dockerfile
parse error: Invalid numeric literal at line 2, column 8

Testing mergermarket/cdflow-commands:153...
```

---

```
$ snyk test --docker mergermarket/cdflow-commands:152 --file=Dockerfile --debug
Failed to run the process ...
Error: 
Testing mergermarket/cdflow-commands:152...

Docker error: image was not found locally: mergermarket/cdflow-commands:152
    at /usr/local/lib/node_modules/snyk/src/cli/commands/test.ts:166:19
    at Generator.throw (<anonymous>)
    at rejected (/usr/local/lib/node_modules/snyk/dist/cli/commands/test.js:5:65)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
```

**Comment by grahamlyons** [comment]
This did not happen when using an image in AWS ECR - the first command failed but the image was no downloaded.

**Comment by RotemS** [comment]
Hi @grahamlyons 👋  Wanted to update that this issue is fixed in snyk version `1.297.0`. I'm closing the issue, please re-open if you encounter any issues! Thanks for raising to our attention.
